hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/launcher/ContainerLaunch.java

```diff
@@ -1382,73 +1410,53 @@ public void setExitOnFailure() {
         return Collections.emptySet();
       }
       final Set<String> deps = new HashSet<>();
-      // env/whitelistedEnv dump values inside double quotes
-      boolean inDoubleQuotes = true;
-      char c;
-      int i = 0;
-      final int len = envVal.length();
-      while (i < len) {
-        c = envVal.charAt(i);
-        if (c == '"') {
-          inDoubleQuotes = !inDoubleQuotes;
-        } else if (c == '\'' && !inDoubleQuotes) {
-          i++;
-          // eat until closing simple quote
-          while (i < len) {
-            c = envVal.charAt(i);
-            if (c == '\\') {
-              i++;
-            }
-            if (c == '\'') {
-              break;
-            }
-            i++;
+      fillDeps(deps, envVal);
+      return deps;
+    }
+
+    private void fillDeps(final Set<String> deps, final String input) {
```

Comment:
I think comments for the three parts (variable name matching, double quote matching, shell expansion matching) would be necessary for future reference. Perhaps an example would be helpful as well for each part.

Reply:
Added some explanation in comments.